tools/mo/openvino/tools/mo/utils/ir_reader/layer_to_class.py

```diff
@@ -195,12 +195,13 @@ def groupconv_to_conv(op: Node):
         weights_node.value = np.reshape(weights_node.value, new_shape)
     elif weights_node.type == 'Reshape':
         # We remove reshape node added in ConvolutionWithGroupsResolver pass
-        assert weights_node.in_port(0).get_source().data.get_shape() == new_shape, \
+        assert strict_compare_tensors(weights_node.in_port(0).get_source().data.get_shape(), new_shape), \
             'Weight shape and calculated shape mismatch in GroupConv node {}.'.format(op.name)
         op.in_port(1).disconnect()
         # We use add_destination method here to support case with multiple destinations of source port
         weights_node.in_port(0).get_source().get_connection().add_destination(op.in_port(1))
         weights_node.in_port(0).disconnect()
+        op.graph.remove_node(weights_node.id)
```

Comment:
Is this really needed? Why can't this node be removed by graph.clean_up() at later stage?